src/Illuminate/Support/Collection.php

```diff
@@ -1548,9 +1548,9 @@ public function offsetSet($key, $value)
     {
         if (is_null($key)) {
             $this->items[] = $value;
-        } else {
-            $this->items[$key] = $value;
         }
+
+        $this->items[$key] = $value;
```

Comment:
you broke the functionality of at least 1 method for no reason at all ... what do you think is going to happen? :)